lib/sphinx/integration/extensions/riddle/client.rb

```diff
@@ -40,52 +40,60 @@ def connect_with_pooling
+                  end
+                when :status
+                  socket.send request_header(command, message.length), 0
                 else
-                  socket.send request_header(command, message.length) +
-                    [messages.length].pack('N') + message, 0
+                  socket.send request_header(command, message.length) + message, 0
                 end
-              when :status
-                socket.send request_header(command, message.length), 0
-              else
-                socket.send request_header(command, message.length) + message, 0
-              end
-
-              # Receive header
-              header = ''.dup
-              if read_timeout.nil?
-                header = socket.recv(HEADER_LENGTH)
-              else
-                nonblock_read!(socket, HEADER_LENGTH, header, read_timeout)
-              end
-              status, version, length = header.unpack('n2N')
 
-              # Если вернулся ответ Retry, то нужно попробовать
-              # отправить запрос на другую ноду, если имеется таковая
-              # дальнейшее чтение из сокета ненужно, закроем его выше
-              if ::Riddle::Client::Statuses[:retry] == status
-                raise ::Riddle::ResponseError, 'Searchd responded with retry error'
-              end
+                # Receive header
+                if read_timeout.nil?
+                  header = socket.recv(HEADER_LENGTH)
+                else
+                  nonblock_read!(socket, HEADER_LENGTH, header, read_timeout)
+                end
+                status, version, length = header.unpack('n2N')
+                length = length.to_i
+
+                # Если вернулся ответ Retry, то нужно попробовать
+                # отправить запрос на другую ноду, если имеется таковая
+                # дальнейшее чтение из сокета ненужно, закроем его выше
+                if ::Riddle::Client::Statuses[:retry] == status
+                  raise ::Riddle::ResponseError, 'Searchd responded with retry error'
+                end
 
-              # Receive body
-              if read_timeout.nil?
-                block_read(socket, length, response)
-              else
-                nonblock_read!(socket, length, response, read_timeout)
-              end
+                # Receive body
+                if read_timeout.nil?
+                  block_read(socket, length, response)
+                else
+                  nonblock_read!(socket, length, response, read_timeout)
+                end
 
-              if response.empty? || response.bytesize != length
-                raise ::Riddle::ResponseError, "No response from searchd (status: #{status}, version: #{version})"
+                if response.empty? || response.bytesize != length
+                  raise ::Riddle::ResponseError, "No response from searchd (status: #{status}, version: #{version})"
+                end
+              rescue Exception
```

Comment:
Lint/RescueException: Avoid rescuing the Exception class. Perhaps you meant to rescue StandardError?